compiler/src/dotty/tools/dotc/core/TypeComparer.scala

```diff
@@ -2005,10 +2005,15 @@ class TypeComparer(initctx: Context) extends ConstraintHandling[AbsentContext] {
         intersecting(tp1.tp1, tp2) || intersecting(tp1.tp2, tp2)
       case (_, tp2: OrType)  =>
         intersecting(tp1, tp2.tp1) || intersecting(tp1, tp2.tp2)
+      case (tp1: AndType, tp2: AndType) =>
+        intersecting(tp1.tp1, tp2.tp1) && intersecting(tp1.tp2, tp2.tp2) ||
+        intersecting(tp1.tp1, tp2.tp2) && intersecting(tp1.tp2, tp2.tp1)
       case (tp1: AndType, _) =>
-        intersecting(tp1.tp1, tp2) && intersecting(tp1.tp2, tp2) && intersecting(tp1.tp1, tp1.tp2)
+        intersecting(tp1.tp1, tp2) && intersecting(tp1.tp2, tp2) ||
```

Comment:
That looks like the same thing is tested twice (???)

Reply:
I don't understand the intersecting logic.

Reply:
Given that you have only certainty for things that do _not_ intersect, would it be better to flip the method and call it `disjoint`? It feels more naturally if unknown is treated as false.

Reply:
I renamed the method to `disjoint` and flipped all the logic, I think it's more natural that way!